### PR TITLE
WIP: Preload browser differences (for discussion, not for merging)

### PR DESCRIPTION
**Pull request by noamr**
This test is meant to support the discussion in
https://github.com/whatwg/fetch/issues/590.

The test accesses a URL that returns an integer that gets incremented
with each request (starting with 0).

The test preloads it once, and then loads it twice.

In Firefox, the test returns 0,0
In Chrome, the test returns 0,1 (preload is used for one request)
In Safari, the test returns 1,2 (preload makes an unused request)

### CODE REVIEW

File: preload/preload-edge-cases.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        const response = await fetch(`${serveUrl}?key=${key}`, {cache: 'force-cache'})
+        return await response.text()
+    }
+
+    function preload(key) {
+        const link = document.createElement('link')
+        link.rel = "preload"
+        link.href = `${serveUrl}?key=${key}`
+        link.as = "fetch"
+        link.setAttribute("crossorigin", "crossorigin")
```
Comment (yoavweiss) [inline preload/preload-edge-cases.html:23]: The value should be "anonymous" or ""

File: preload/preload-edge-cases.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE html>
+<title>Test cases for scenarios not defined in the preload spec</title>
+<script src="/resources/testharness.js" a></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+    const serveUrl = './resources/inc.py'
+    function uuidv4() {
+        return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+            (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
```
Comment (yoavweiss) [inline preload/preload-edge-cases.html:9]: https://wicg.github.io/uuid/ could've been interesting here, but I don't think it's broadly supported.

**Comment by noamr** [comment]
I updated the test cases to also include script and image loading, and to try several combinations of cache headers from the server (can possibly add more).
Also I'm separating the count from the response content, to make it easier to add new resource types in the future.

^^ @pmeenan
